tf/do/main.tf

```diff
@@ -0,0 +1,220 @@
+    #     }
+    # }
+
+    # provisioner "file" {
+    #     source      = "files/rclone.sh"
+    #     destination = "/home/ec2-user/rclone.sh"
+    #     connection {
+    #     host        = "${aws_spot_instance_request.foundation-node.public_ip}"
+    #     type        = "ssh"
+    #     user        = "ec2-user"
+    #     private_key = "${file(var.private_key_path)}"
+    #     agent       = true
+    #     }
+    # }
+
+    # provisioner "file" {
+    #     source      = "files/uploadlog.sh"
+    #     destination = "/home/ec2-user/uploadlog.sh"
+    #     connection {
+    #     host        = "${aws_spot_instance_request.foundation-node.public_ip}"
+    #     type        = "ssh"
+    #     user        = "ec2-user"
+    #     private_key = "${file(var.private_key_path)}"
+    #     agent       = true
+    #     }
+    # }
+
+    # provisioner "file" {
+    #     source      = "files/crontab"
+    #     destination = "/home/ec2-user/crontab"
+    #     connection {
+    #     host        = "${aws_spot_instance_request.foundation-node.public_ip}"
+    #     type        = "ssh"
+    #     user        = "ec2-user"
+    #     private_key = "${file(var.private_key_path)}"
+    #     agent       = true
+    #     }
+    # }
+
+
+    # provisioner "remote-exec" {
+    #     inline = [
+    #     "curl -LO https://harmony.one/node.sh",
+    #     "chmod +x node.sh rclone.sh fast.sh uploadlog.sh",
+    #     "crontab crontab",
+    #     "mkdir -p /home/ec2-user/.config/rclone",
+    #     "mv -f rclone.conf /home/ec2-user/.config/rclone",
+    #     "sudo mv -f harmony.service /etc/systemd/system/harmony.service",
+    #     "sudo systemctl enable harmony.service",
+    #     "sudo systemctl start harmony.service",
+    #     "sudo mv -f node_exporter.service /etc/systemd/system/node_exporter.service",
+    #     "sudo systemctl daemon-reload",
+    #     "sudo systemctl start node_exporter",
+    #     "sudo systemctl enable node_exporter",
+    #     "echo ${var.blskey_index} > index.txt",
+    #     ]
+    #     connection {
+    #     host        = "${aws_spot_instance_request.foundation-node.public_ip}"
+    #     type        = "ssh"
+    #     user        = "ec2-user"
+    #     private_key = "${file(var.private_key_path)}"
+    #     agent       = true
+    #     }
+    # } 
+
+}
+
+resource "digitalocean_volume" "harmony_data_volume" {
+    region                  = var.droplet_region
+    name                    = "volume-s${var.shard}-${var.blskey_index}-${random_id.droplet_id.hex}"
+    size                    = var.addition_volume_size
+    initial_filesystem_type = "ext4"
+    description             = "the data volume of harmony foundational node"
+}
+
+resource "digitalocean_firewall" "harmony_fw" {
+    name = "harmony-fw"
+
+    droplet_ids = [digitalocean_droplet.harmony_node.id]
+
+    inbound_rule {
+        protocol         = "tcp"
+        port_range       = "22"
+        source_addresses = ["0.0.0.0/0", "::/0"]
```

Comment:
please don't open it general public. SSH has to be whitelisted.

Reply:
fixed